pkg/engine/mutation.go

```diff
@@ -144,10 +144,21 @@ func Mutate(policyContext *PolicyContext) (resp *response.EngineResponse) {
 }
 
 func mutateResource(rule *kyvernov1.Rule, ctx *PolicyContext, resource unstructured.Unstructured, logger logr.Logger) (*response.RuleResponse, unstructured.Unstructured) {
+	// checks preconditions under rules block
 	preconditionsPassed, err := checkPreconditions(logger, ctx, rule.GetAnyAllConditions())
 	if err != nil {
 		return ruleError(rule, response.Mutation, "failed to evaluate preconditions", err), resource
 	}
+	// Check preconditions under rule.Mutation block
+	if !ctx.AdmissionOperation && rule.IsMutateExisting() {
```

Comment:
Can we move this into `mutate.Mutate()` method as it checks conditions explicitly defined in the mutate overlay?